src/binder/bind/bind_updating_clause.cpp

```diff
@@ -152,7 +153,10 @@ static void validatePrimaryKeyExistence(const NodeTableCatalogEntry* nodeTableEn
         }
         return; // No input needed for SERIAL primary key.
     }
-    if (!node.hasPropertyDataExpr(primaryKey->getName())) {
+    auto pkeyDefault = defaultExprs.at(nodeTableEntry->getPrimaryKeyPos());
+    if (!node.hasPropertyDataExpr(primaryKey->getName()) &&
+        pkeyDefault->expressionType == ExpressionType::LITERAL &&
```

Comment:
I would add a function in ExpressionUtil to check if a given expression is NULL literal or not